tests/test_core/test_generic_section.py

```diff
@@ -128,6 +128,382 @@ def test_rectangular_section():
+    sec = GenericSection(geo, integrator='fiber', mesh_size=0.0001)
+    assert sec.name == 'GenericSection'
+
+    assert math.isclose(sec.gross_properties.area, b * h)
+
+    # compute stiffness matrix
+    stiffness = sec.section_calculator.integrate_strain_profile(
+        [0, 0, 0], 'modulus'
+    )
+    assert stiffness.shape == (3, 3)
+    stiffness /= E
+
+    # Create the section with marin integrator
+    sec = GenericSection(geo)
+    assert sec.name == 'GenericSection'
+
+    assert math.isclose(sec.gross_properties.area, b * h)
+
+    # compute stiffness matrix
+    stiffness = sec.section_calculator.integrate_strain_profile(
+        [0, 0, 0], 'modulus'
+    )
+    assert stiffness.shape == (3, 3)
+    stiffness /= E
+
+    # compute stiffness matrix analitically
+    # The first moment and product moment should be zero in this case
+    stiffness_expected = np.zeros((3, 3))
+    stiffness_expected[0, 0] = b * h
+    stiffness_expected[1, 1] = 1 / 12.0 * b * h**3
+    stiffness_expected[2, 2] = 1 / 12.0 * h * b**3
+    assert np.allclose(
+        stiffness,
+        stiffness_expected,
+        atol=stiffness.max() * 1e-6,
+        rtol=1e-2,
+    )
+
+
+# Test rectangular RC section tangent stiffness initial
+@pytest.mark.parametrize(
+    'b, h, c, diameter, n_bottom, n_top',
+    [
+        (250, 500, 40, 20, 3, 2),
+    ],
+)
+def test_rectangular_rc_section_initial_tangent_stiffness(
+    b, h, c, diameter, n_bottom, n_top
+):
+    """Test stiffness matrix of RC rectangular section."""
+    # Create materials to use
+    concrete = ConcreteEC2_2004(30)
+    Es = 200000
+    steel = ReinforcementEC2_2004(
+        450, Es=Es, ftk=450, epsuk=0.075, gamma_s=1.15, gamma_eps=0.9
+    )
+
+    # The section
+    poly = Polygon(
+        ((-b / 2, -h / 2), (b / 2, -h / 2), (b / 2, h / 2), (-b / 2, h / 2))
+    )
+    geo = SurfaceGeometry(poly, concrete)
+
+    assert geo.polygon.centroid.coords[0][0] == 0
+    assert geo.polygon.centroid.coords[0][1] == 0
+
+    # Add reinforcement
+    geo = add_reinforcement_line(
+        geo,
+        coords_i=(-b / 2 + c, -h / 2 + c),
+        coords_j=(b / 2 - c, -h / 2 + c),
+        diameter=diameter,
+        material=steel,
+        n=n_bottom,
+    )
+    geo = add_reinforcement_line(
+        geo,
+        coords_i=(-b / 2 + c, h / 2 - c),
+        coords_j=(b / 2 - c, h / 2 - c),
+        diameter=diameter,
+        material=steel,
+        n=n_top,
+    )
+
+    # Manual computations
+    Ec = concrete.constitutive_law.get_tangent(0)
+    EA_gross = Ec * b * h
+    EIyy_gross = Ec * 1 / 12.0 * b * h**3
+    EIzz_gross = Ec * 1 / 12.0 * h * b**3
+    for p in geo.point_geometries:
+        y, z = p.point.coords.xy
+        As = p.area
+        EA_gross += Es * As
+        EIyy_gross += Es * As * z[0] ** 2
+        EIzz_gross += Es * As * y[0] ** 2
+
+    # Create the section with fiber integrator
+    sec = GenericSection(geo, integrator='fiber', mesh_size=0.0001)
```

Comment:
Same here, could we parameterize the name of the integrator?

Reply:
sure!